### ISSUE TITLE
Implement weapon

### ISSUE THREAD

**Issue by johnthagen** [johnthagen/rust-belt#5]
One idea is that`rocket` uses a `World` to keep track of all of the different entities that exist at any moment: https://github.com/aochagavia/rocket/blob/master/src/models/world.rs#L9

**Comment by mcdenhoed** [comment]
Check out the weapon branch. A very rudimentary implementation is sitting there. Didn't feel like learning to draw non-triangles, so the bullets are triangles.

Bullets should probably be tied to the player object somehow instead of just sitting on their own. They should also inherit the player's velocity, and maybe provide recoil.

**Comment by johnthagen** [comment]
Very cool! I think the actual parameters (speed, ttl, etc) can be tweaked later but I think the overall approach looks great. I think how it fits into the new reorganization and how natural the `Vec<bullet::Bullet>` feels.

> so the bullets are triangles.

To draw circles, you can use the [`ellipse`](http://docs.piston.rs/piston_window/graphics/fn.ellipse.html) function.

Here is an example from rocket:

https://github.com/aochagavia/rocket/blob/master/src/models/bullet.rs

```rust
    /// Draw the bullet
    pub fn draw(&self, c: &Context, gl: &mut GlGraphics) {
        ellipse(color::BLUE,
                [self.x() - self.radius(),
                    self.y() - self.radius(),
                    self.diameter(),
                    self.diameter()],
                c.transform,
                gl);
    }
```

In `bullet::update` should the `ttl` be based on `args.dt`? This would also have the side effect of getting to remove the `#[allow(unused_variables)]`.

in `Bullet::new`, `Size` implements `Copy` and is a value-type, so let's just pass a copy.

```rust
... window_size: Size)
```

~~In `struct Bullet` does `ttl` need to be `pub`?~~

> Bullets should probably be tied to the player object somehow instead of just sitting on their own. 

I think it's okay for now to leave them independent. I could imagine a simple game where they are independent so I think we can keep things simple for now.

**Comment by mcdenhoed** [comment]
@johnthagen Pushed the changes you suggested to weapon

**Comment by johnthagen** [comment]
Nice! I like the new look of the bullets.

Couple ideas:

- [x]  If we draw `Bullet`s before the `Player` then they would look like they came from inside the player's ship. Or we could start the bullet from the tip of the player's ship, but that's more complex.
- [x] Can remove `#[allow(unused_variables)]` from `Bullet::update()`
- [x]  `self.bullets.retain(|x| x.ttl() > 0.0);` would this be a little more clear if we used `b` or `bullet` for the arbitrary variable name in the closure rather than `x`? 
- [x] In order to turn and shoot at the same time, do we need to `push` the `Bullet` during the `Update` cycle instead of the `Input` cycle? I think this is how we fixed this issue for turning.
- [x] Module docs for `bullet.rs` need to be updated.

And a more general question, we have a lot of `const`s that really are only used inside a single function (`BULLET`, `SPEED_MULTIPLIER`, `SHIP`, etc). I think it can be a separate PR, but if they don't need to be in the module scope, it seems cleaner to just define them in the single function they are used in. This is how `story::render()` does it.
